client/py/tests/test_framer.py

```diff
@@ -92,6 +93,55 @@ def test_write_frame_idx_no_timestamps(
                 data = np.random.rand(10).astype(np.float64)
                 w.write(pd.DataFrame({data_ch.key: data}))
 
+    def test_write_auto_commit(self, channel: sy.Channel, client: sy.synnax):
+        """Should open an auto-committing writer to write data that persists after 1s"""
+        with client.open_writer(0, channel.key, enable_auto_commit=True) as w:
+            data = np.random.rand(10).astype(np.float64)
+            w.write(pd.DataFrame({channel.key: data}))
+            w.write(pd.DataFrame({channel.key: data}))
+            assert w.error() is None
+
+        f = client.read(TimeRange(0, TimeStamp(1 * TimeSpan.SECOND)), channel.key)
+        assert f.__len__() == 20
+
+    def test_write_auto_commit_always_persist(self, channel: sy.Channel, client: sy.Synnax):
+        """Should open an auto-committing writer to write data to Synnax."""
+        with client.open_writer(0, channel.key, enable_auto_commit=True,
+                                auto_index_persist_interval=
+                                sy.Writer.ALWAYS_INDEX_PERSIST_ON_AUTO_COMMIT
+                                ) as w:
+            data = np.random.rand(10).astype(np.float64)
+            w.write(pd.DataFrame({channel.key: data}))
+            w.write(pd.DataFrame({channel.key: data}))
+            assert w.error() is None
+
+        f = client.read(TimeRange(0, TimeStamp(1 * TimeSpan.SECOND)), channel.key)
+        assert f.__len__() == 20
+
+    def test_write_auto_commit_set_persist(self, channel: sy.Channel, client: sy.Synnax):
+        """Should open an auto-committing-and-persisting writer to write data."""
+        with client.open_writer(0,
+                                channel.key,
+                                enable_auto_commit=True,
+                                auto_index_persist_interval=50 * TimeSpan.MILLISECOND) as w:
+            data = np.random.rand(10).astype(np.float64)
+            w.write(pd.DataFrame({channel.key: data}))
+            w.write(pd.DataFrame({channel.key: data}))
+
+        f = client.read(TimeRange(0, TimeStamp(1 * TimeSpan.SECOND)), channel.key)
+        assert f.__len__() == 20
+
+    def test_writer_creation_with_bad_params(self, channel: sy.Channel, client: sy.Synnax):
```

Comment:
I don't know if these should actually be error cases. Can't it just be that any negative number means always persist? And can't we just ignore the auto persist interval if auto commit is off?

Reply:
Agreed